torchvision/prototype/models/alexnet.py

```diff
@@ -29,16 +32,10 @@ class AlexNetWeights(Weights):
     )
 
 
-def alexnet(weights: Optional[AlexNetWeights] = None, progress: bool = True, **kwargs: Any) -> AlexNet:
-    if type(weights) == bool and weights:
-        _deprecated_positional(kwargs, "pretrained", "weights", True)
-    if "pretrained" in kwargs:
-        weights = _deprecated_param(kwargs, "pretrained", "weights", AlexNetWeights.ImageNet1K_RefV1)
+@handle_legacy_interface(AlexNetWeights.ImageNet1K_RefV1)
+@handle_num_categories_mismatch()
+def alexnet(*, weights: Optional[AlexNetWeights] = None, progress: bool = True, **kwargs: Any) -> AlexNet:
     weights = AlexNetWeights.verify(weights)
-
-    if weights is not None:
-        _ovewrite_named_param(kwargs, "num_classes", len(weights.meta["categories"]))
```

Comment:
Could you explain your thinking behind this? IIUC, this is input validation and if multiple functions need the same, a decorator fits well. That way the body of the builder becomes more readable because it can focus on on the actual functionality.

Reply:
Sure. This is not input validation but affects the model config which is key part of the method builder. This is also where the default value of the parameter lives (because the signature defines it as `None`). The idiom of modifying the model configuration prior calling the model class is very common across TorchVision and belongs in the method builder. You can see similar examples on the InceptionV3 builder and on the `_validate_trainable_layers()` calls of all detection models. Similarly `num_classes`, `num_keypoints` etc are building parameters of the model classes and changing their configuration should remain inside the model builder.